lib/bundler/source.rb

```diff
@@ -38,5 +38,17 @@ def include?(other)
     def inspect
       "#<#{self.class}:0x#{object_id} #{self}>"
     end
+
+  private
+
+    def version_color(spec_version, locked_spec_version)
+      earlier_version?(spec_version, locked_spec_version) ? :yellow : :green
+    end
+
+    def earlier_version?(spec_version, locked_spec_version)
+      Gem::Version.new(spec_version) < Gem::Version.new(locked_spec_version)
+    rescue ArgumentError
```

Comment:
Why would these be dependencies as opposed to versions? Since the parameter name is `version` -- if it's not possible to get the actual version, I'd prefer we check `Gem::Version.correct?` instead of using exceptions for control flow